### PR TITLE
Added `params` and `query` to `useRouter` and `withRouter`

### PR DESCRIPTION
**Pull request by toshi1127**
This PR closes the issue of Add `params` and `query` to `useRouter` and `withRouter`. #623 

### What are the changes and their implications?
Extend useRouter and withRouter with useParams and useRouterQuery hooks.

### Current:
```
const {query} = useRouter()
```

### Change to:

```
const {params, query} = useRouter()
```

### Checklist

- [ ] Tests added for changes
- [x] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes: https://github.com/blitz-js/blitzjs.com/pull/82

### CODE REVIEW

**Comment by toshi1127** [comment]
Should update the [documentation](https://blitzjs.com/docs/router#userouter) and implement the test code in this PR?

File: packages/core/src/use-router.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {useRouter as useRouterByNext} from 'next/router'
```
Comment (Zeko369) [inline packages/core/src/use-router.ts:1]: Maybe rename to `useNextRouter` or `useRouterNext`, `by` seems weird IMO

Comment (toshi1127) [reply packages/core/src/use-router.ts]: Renamed to useRouterNext.

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -23,8 +25,6 @@ export {default as Head} from 'next/head'
 
 export {default as Link} from 'next/link'
 
-export {Router, useRouter, withRouter} from 'next/router'
```
Comment (flybayer) [inline packages/core/src/index.ts:26]: We still need to export `Router` from `next/router`

File: packages/core/src/with-router.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import {withRouter as withRouterNext, NextRouter as NextRouterNext} from 'next/router'
```
Comment (flybayer) [inline packages/core/src/with-router.tsx:2]: You only need 'next' one time in `NextRouterNext` :D

File: packages/core/src/index.ts (hunk 2 above)
Comment (toshi1127) [reply packages/core/src/index.ts:26]: Fixed.

File: packages/core/src/with-router.tsx (hunk 3 above)
Comment (toshi1127) [reply packages/core/src/with-router.tsx]: Admittedly, this renaming is unnecessary.
Fixed.

**Comment by toshi1127** [comment]
Thanks for the review !!
Admittedly, given the grammar, useNextRouter, withNextRouter is appropriate.
These have been fixed.

**Comment by flybayer** [comment]
@toshi1127 awesome, looks great!

One last thing. Can you open a PR to update [the router object documentation](https://blitzjs.com/docs/router) for this? [Here's the file on Github](https://github.com/blitz-js/blitzjs.com/edit/master/docs/router.mdx)

**Comment by toshi1127** [comment]
@flybayer 
Thank you.
Of course I can update the document.
I'll create a PR when I'm done updating it.

**Comment by toshi1127** [comment]
Created a [PR](https://github.com/blitz-js/blitzjs.com/pull/82) to update the documentation of the router object.

**Comment by flybayer** [comment]
Great, thank you! :)

**Comment by flybayer** [comment]
@all-contributors add @toshi1127 for docs

**Pull request merged**